src/main/scala/org/constellation/domain/transaction/TransactionService.scala

```diff
@@ -57,50 +57,77 @@ class TransactionService[F[_]: Concurrent](val transactionChainService: Transact
   def createTransaction(
     src: String,
     dst: String,
+    prevTxRef: String,
+    ordinal: Long,
     amount: Long,
     keyPair: KeyPair,
     normalized: Boolean = true,
     dummy: Boolean = false
-  ): F[Transaction] =
-    TransactionService.createTransaction(src, dst, amount, keyPair, normalized, dummy)(transactionChainService)
+  ): F[Transaction] = TransactionService.createTransaction(src, dst, prevTxRef, ordinal, amount, keyPair, normalized, dummy)(transactionChainService)
 
-  def createDummyTransaction(src: String, dst: String, keyPair: KeyPair): F[Transaction] =
-    TransactionService.createDummyTransaction(src, dst, keyPair)(transactionChainService)
+  def createDummyTransaction(src: String, dst: String, keyPair: KeyPair, prevTxRef: String = "dummy", ordinal: Long = 0L): F[Transaction] =
```

Comment:
Why dummy transaction will not have any reference? The dummy transaction is kinda real anyway, so we should keep track of tx chain for those addresses, etc 🤔